docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -0,0 +1,350 @@
+# Machine Learning Inference C-API
+
+Developers can easily construct and efficiently execute data stream pipelines consisting of neural networks as filters in pipelines.
+
+The main features of Machine Learning Inference C-API include:
+
+- **Construction of data pipeline based on [gstreamer](https://gstreamer.freedesktop.org/)**
+
+  You can compose the data stream pipeline through Machine Learning Inference C-API with various elements of gstreamer and nnstreamer.
+
+- **[Single](#singleapi) & [Pipeline](#pipelineapi) API**
+  
+  There are 2 types of Machine Learning Inference C-API. Single API is useful for simple usage scenario of neural network models. It allows to invoke a neural network model with a single instance of input data for the model directly. It is useful if you have the input data pre-processed with the application itself and there is no complex interactions between neural network models, data processors, or data stream paths.
+  
+  On the other hand, Pipeline API allows developers to construct and execute pipelines with multiple neural network models, multiple input and output nodes, multiple data processors, pre-and-post processors, and various data path manipulators. Besides, if the input is online data (or streamed data), Pipeline APIs simplify your application and improve its performance.
+  
+- **Support various neural net frameworks (NNFW)**
+  
+  TensorFlow, TensorFlow-Lite, Caffe2, and PyTorch are support. Thus, neural network model files trained by such frameworks can be imported as filters of pipelines directly.
+  Custom filters, which are neural network models implemented directly with programming languages including C/C++ and Python, may be imported as filters of pipelines directly as well.
+
+  Note that, in Tizen release, this is limited to TensorFlow-Lite only; we do not ship other frameworks for official embedded device OS images.
+
+## Prerequisite <a name="prerequisite"></a>
+
+To use the APIs, below headers and features are required.
+
+- **Header**
+  ```bash
+  <nnstreamer.h>
+  ```
+
+- **Feature**
+  ```bash
+  <feature name="http://tizen.org/feature/machine_learning">true</feature>
+  <feature name="http://tizen.org/feature/machine_learning.inference">true</feature>
+  ```
+
+
+## Single API <a name="singleapi"></a>
```

Comment:
As a best practice, it would be good to give a small introduction line before the steps begin.